src/services/AnalysisService.ts

```diff
@@ -0,0 +1,326 @@
+  projectName: string;
+  branchName: string;
+  commitHash: string;
+  buildVersion: string;
+  buildUri: string;
+  branchUri: string;
+  integrationType: string;
+  operatingEnvironment: string;
+  integrationName: IntegrationName;
+  appVersion: string;
+  scriptVersion: string;
+  contributingDeveloperAudit: ICreateScanRequestContributingDeveloperAudit[];
+  scanType: ScanType;
+  toolName?: string;
+  toolVersion?: string;
+}
+
+interface IUpdateScanStatusParams {
+  clientId: string;
+  projectHash: string;
+  branchHash: string;
+  scanType: ScanType;
+  analysisId: string;
+  status: ScanStatus;
+  message: string;
+}
+
+const integrationNameToEnvVariable: Record<IntegrationName, string> = {
+  [IntegrationName.AWSCodeBuild]: "CODEBUILD_BUILD_INITIATOR",
+  [IntegrationName.Bamboo]: "bamboo_planRepository_1_username",
+  [IntegrationName.BitBucket]: "BITBUCKET_STEP_TRIGGERER_UUID",
+  [IntegrationName.CircleCI]: "CIRCLE_USERNAME",
+  [IntegrationName.CodeShip]: "CI_COMMITTER_USERNAME",
+  [IntegrationName.GithubActions]: "GITHUB_ACTOR",
+  [IntegrationName.GitLab]: "GITLAB_USER_LOGIN",
+  [IntegrationName.Jenkins]: "CHANGE_AUTHOR",
+  [IntegrationName.Script]: "SOOS_CONTRIBUTING_DEVELOPER",
+  [IntegrationName.TeamCity]: "TEAMCITY_BUILD_TRIGGEREDBY_USERNAME",
+  [IntegrationName.TravisCI]: "TRAVIS_COMMIT",
+};
+
+class AnalysisService {
+  public analysisApiClient: SOOSAnalysisApiClient;
+  public projectsApiClient: SOOSProjectsApiClient;
+  public usersApiClient: SOOSUsersApiClient;
+
+  constructor(
+    analysisApiClient: SOOSAnalysisApiClient,
+    projectsApiClient: SOOSProjectsApiClient,
+    usersApiClient: SOOSUsersApiClient,
+  ) {
+    this.analysisApiClient = analysisApiClient;
+    this.projectsApiClient = projectsApiClient;
+    this.usersApiClient = usersApiClient;
+  }
+
+  static create(apiKey: string, apiURL: string): AnalysisService {
+    const analysisApiClient = new SOOSAnalysisApiClient(apiKey, apiURL);
+    const projectsApiClient = new SOOSProjectsApiClient(
+      apiKey,
+      apiURL.replace("api.", "api-projects."),
+    );
+    const usersApiClient = new SOOSUsersApiClient(apiKey, apiURL.replace("api.", "api-user."));
+
+    return new AnalysisService(analysisApiClient, projectsApiClient, usersApiClient);
+  }
+
+  async setupScan({
+    clientId,
+    projectName,
+    branchName,
+    commitHash,
+    buildVersion,
+    buildUri,
+    branchUri,
+    integrationType,
+    operatingEnvironment,
+    integrationName,
+    appVersion,
+    scriptVersion,
+    contributingDeveloperAudit,
+    scanType,
+    toolName,
+    toolVersion,
+  }: ISetupScanParams): Promise<ICreateScanResponse> {
+    soosLogger.info("Getting application status...");
+    const applicationStatus = await this.usersApiClient.getApplicationStatus(clientId);
+    if (applicationStatus.statusMessage) {
+      soosLogger.info(applicationStatus.statusMessage.message);
+      if (applicationStatus.statusMessage.linkText) {
```

Comment:
there may not be a link text - check the url field here, and only show the text if it is there too